src/pugixml.hpp

```diff
@@ -38,6 +38,20 @@
 #	include <string>
 #endif
 
+// Check if std::string_view is both requested and available
+#if defined(PUGIXML_STRING_VIEW) && !defined(PUGIXML_NO_STL)
+#	if __cplusplus >= 201703L
+#		define PUGI_HAS_STRING_VIEW
```

Comment:
There was some confusion in the original PR wrt naming; this should be `PUGIXML_HAS_STRING_VIEW` as `PUGI_` prefix is not actually used except for defines internal to `pugixml.cpp`

Reply:
Sound good, I will rename. It occurred to me after sending the PR that this check is not too complicated -- if we wanted to avoid leaking the new define, we could copy-paste this check into the .cpp and undefine at the end of the header and .cpp files. Leaving it as is also works, as there is already precedent for the header adding PUGIXML_ defines.

Reply:
I think it's fine to keep the define for now since as you mention there's already other similar defines, and some are also gated on C++ version.